wallet-test/src/test/scala/org/bitcoins/wallet/WalletUnitTest.scala

```diff
@@ -136,16 +136,26 @@ class WalletUnitTest extends BitcoinSWalletTest {
 
   it should "fail to unlock the wallet with a bad password" in {
     wallet: Wallet =>
-      val badpassphrase = AesPassword.fromNonEmptyString("bad")
+      val badPassphrase = Some(AesPassword.fromNonEmptyString("bad"))
 
-      val errorType = wallet.unlock(badpassphrase, None) match {
+      val errorType = wallet.unlock(badPassphrase, None) match {
         case Right(_)  => fail("Unlocked wallet with bad password!")
         case Left(err) => err
       }
       errorType match {
-        case KeyManagerUnlockError.MnemonicNotFound          => fail(MnemonicNotFound)
-        case KeyManagerUnlockError.BadPassword               => succeed
-        case KeyManagerUnlockError.JsonParsingError(message) => fail(message)
+        case KeyManagerUnlockError.MnemonicNotFound => fail(MnemonicNotFound)
+        case KeyManagerUnlockError.BadPassword      =>
+          // If wallet is unencrypted then we shouldn't get a bad password error
```

Comment:
Ok this comment is confusing to me, "we shouldn't get a bad password error" but the pattern match this is part of indicates there was a bad password? I.e. `KeyManagerUnlockError.BadPassword`

Reply:
We shouldn't get a `BadPassword` error when the wallet doesn't use an aes password. But if we do have an aes password then we should get the error, the match on the `aesPasswordOpt` handles this